pkg/utils/nodeclaim/suite_test.go

```diff
@@ -131,4 +136,90 @@ var _ = Describe("NodeClaimUtils", func() {
 			BlockOwnerDeletion: lo.ToPtr(true),
 		}))
 	})
+	Context("List", func() {
```

Comment:
Do we just want to do away with Listing based on providerID since we said that it's an edge case at this point. Fine with a fast-follow though

Reply:
It's still used to get a NodeClaim for a given Node, we're just going to error when there are multiple values going forwards.